exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/columns/ColumnsScanFramework.java

```diff
@@ -103,7 +103,7 @@ protected void configure() {
   }
 
   @Override
-  protected SchemaNegotiatorImpl newNegotiator() {
+  public SchemaNegotiatorImpl newNegotiator() {
```

Comment:
Why is this now `public`? The `columns` framework was for CSV files using the `columns` column; this is an internal bit of structure not meant to be overridden.

Also, would be better to invest time with the "new" version rather than this somewhat cumbersome older version.

Reply:
As it turns out, cleaning up this kind of issue is one of the things that the new version does.

There is also a race condition. The schema negotiator is passed into the reader; it is not meant to be obtained elsewhere. In general, the framework does a bunch of initialization, getting things ready. Then, each reader is created, one at a time, passing in the schema negotiator. If we have code trying to access it at some other time, then we'd have to review all that code to see if new pattern works.

Reply:
Took this into account with new method:
```
    /**
     * @return FileScanFramework or empty object in case it is not binded yet with {@link #bind(ManagedScanFramework)}
     */
    protected Optional<FileScanFramework> fileFramework() {
      return Optional.ofNullable(fileFramework);
    }
```
and
```
It is not possible to detect file format, because the File Framework is not initialized. " +
                "Trying to use default PCAP format plugin to read the file
```
It will help for future new version of `FileReaderFactory`